sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -1351,6 +1351,31 @@ object RepartitionByExpression {
   }
 }
 
+/**
+ * This operator used to rebalance the query result output partitions, so that every partition
+ * is of a reasonable size (not too small and not too big). It can take column names as parameters,
+ * and try its best to partition the query result by these columns. If there are skews, Spark will
```

Comment:
``It also try its best to partition the child output by `partitionExpressions` ``